systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaUserUtils.java

```diff
@@ -186,4 +191,32 @@ public static void waitForKafkaUserMappingIntoKafkaResource(String namespace, St
                 return getUserResult.contains(userName);
             });
     }
+
+    public static void modifyKafkaUserPasswordWithNewSecret(String ns, String kafkaUserResourceName, String customSecretSource, String customPassword, ExtensionContext extensionContext) {
+
+        Secret userDefinedSecret = new SecretBuilder()
+            .withNewMetadata()
+                .withName(customSecretSource)
+                .withNamespace(ns)
+            .endMetadata()
+            .addToData("password", customPassword)
+            .build();
+
+        ResourceManager.getInstance().createResourceWithWait(extensionContext, userDefinedSecret);
+
+        KafkaUserResource.replaceUserResourceInSpecificNamespace(kafkaUserResourceName, ku -> {
+
+            ku.getSpec().setAuthentication(
+                new KafkaUserScramSha512ClientAuthenticationBuilder()
+                    .withPassword(
+                        new PasswordBuilder()
+                            .editOrNewValueFrom()
+                                .withNewSecretKeyRef("password", customSecretSource, false)
+                            .endValueFrom()
+                            .build()
+                    )
+                    .build()
+            );
+        }, ns);
```

Comment:
I think after this modification we should at least wait on `KafkaUser` readiness WDYT @im-konge ? I know that in this use case, @henryZrncik waits for RollingUpdate to happen but in other cases...

Reply:
Yeah sure, waiting for readiness would be good.